src/ODM/MongoDB/Types/CurrencyType.php

```diff
@@ -2,10 +2,10 @@
 
 namespace ZFBrasil\DoctrineMoneyModule\ODM\MongoDB\Types;
 
-use Doctrine\ODM\MongoDB\Types\StringType;
+use Doctrine\ODM\MongoDB\Types\Type;
 use Money\Currency;
 
-class CurrencyType extends StringType
+class CurrencyType extends Type
```

Comment:
sry no, this is an falsely behavior Currency isn't a string type, BTW doctrine2 and proxy manager (we used) doesn't automatically cast to string. Not useabale with u proposed __toString